executor.go

```diff
@@ -925,9 +925,6 @@ func getFieldDef(schema Schema, parentType *Object, fieldName string, blockMeta
 		return TypeMetaFieldDef
 	}
 	if fieldName == TypeNameMetaFieldDef.Name {
-		if blockMeta {
-			return BlockedMetaFieldDef
-		}
 		return TypeNameMetaFieldDef
```

Comment:
Add a comment:

// Always allow __typename to be inspected, since this information is used by some client for caching purpose.